mlflow/models/model.py

```diff
@@ -598,6 +598,8 @@ def log(
             ):
                 _logger.warning(_LOG_MODEL_MISSING_SIGNATURE_WARNING)
             flavor.save_model(path=local_path, mlflow_model=mlflow_model, **kwargs)
+            # TODO: can we move saving MLModel file here? This way we can set model_size_bytes
+            #  here for all flavors here, instead of having to implement it in each save_model
```

Comment:
@dbczumar Would love your inputs here!

Reply:
The proposal is to call this method on the `mlflow_model` `Model()` instance within the `save_model()` function in each flavor instead of calling `save()`? 

This method does additional things compared to `save()`. 

I think for abstraction purposes, having an addition to Model().save(), calling a utility function that calculates the file sizes within the target directory that the MLModel is being saved to would perhaps be a bit more appropriate to maintain consistency between model logging and model saving (since logging a flavor will call the save logic within a flavor).

Reply:
(Synced offline earlier) We agreed on it's actually better to keep calling the utility function from each flavor's own `save_model` for better flexibility and extensibility.